controllers/openstack_ansibleee_controller.go

```diff
@@ -92,18 +92,17 @@ func (r *OpenStackAnsibleEEReconciler) Reconcile(ctx context.Context, req ctrl.R
 	// Always patch the instance status when exiting this function so we can
 	// persist any changes.
 	defer func() {
-		ready := true
-		for _, c := range instance.Status.Conditions {
-			if c.Type == condition.ReadyCondition {
-				continue
-			}
-			if c.Status != corev1.ConditionTrue {
-				ready = false
-				break
-			}
+		// update the overall status condition if service is ready
+		if instance.IsReady() {
+			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, redhatcomv1alpha1.AnsibleExecutionJobReadyMessage)
 		}
-		if ready {
-			instance.Status.Conditions.MarkTrue(condition.ReadyCondition, condition.ReadyMessage)
+		c := instance.Status.Conditions.Mirror(redhatcomv1alpha1.AnsibleExecutionJobReadyCondition)
+		if c.Reason == condition.ErrorReason {
+			instance.Status.Conditions.MarkFalse(
+				condition.ReadyCondition,
+				condition.ErrorReason,
+				condition.SeverityError,
+				c.Message)
```

Comment:
Mirror can generate the ReadyCondition directly based on the rest of the conditions:

```golang
if instance.IsReady() {
    instance.Status.Conditions.MarkTrue(condition.ReadyCondition, redhatcomv1alpha1.AnsibleExecutionJobReadyMessage)
} else {
    // something is not ready so reset the Ready condition
    instance.Status.Conditions.MarkUnknown(
        condition.ReadyCondition, condition.InitReason, condition.ReadyInitMessage)
    // and recalculate it based on the state of the rest of the conditions
    instance.Status.Conditions.Set(instance.Status.Conditions.Mirror(condition.ReadyCondition))
}
```